test/testUtils.ts

```diff
@@ -31,11 +31,21 @@ export async function getOptionsForFixture(file: string[]) {
 }
 
 async function getTextEditorOptions() {
+	let resolved = false;
+
 	return new Promise<TextEditorOptions>(resolve => {
 		window.onDidChangeTextEditorOptions(e => {
+			resolved = true;
 			assert.ok(e.options);
 			resolve(e.options);
 		});
+
+		setTimeout(() => {
+			if (!resolved) {
+				assert.ok(window.activeTextEditor.options);
+				resolve(window.activeTextEditor.options);
+			}
+		}, 100);

```

Comment:
Interesting! It seems there might be some value creating a test to validate what you're saying as well.